### ISSUE TITLE
Replace CTD surface measurements with YSI measurements

### ISSUE THREAD

**Issue by Br-Johnson** [HakaiInstitute/data-mgmt#7]

**Comment by Br-Johnson** [comment]
Download ysi data from the portal using the api, and then join surface data from the YSI to the bottle file by event_pk (most likely) into it's own columns for temp_0 and temp_1 and sal_0, sal_1.

**Comment by JessyBarrette** [comment]
I would not replace CTD data with YSI data since those are very different sensor types with a whole different sampling protocol. I would handle the YSI data has a separate variable, kind of like a sample type. 

Regarding the CTD data itself, I would match with the closest available in-depth. For surface samples, getting the shallowest record makes sense to me.

**Comment by jclgan** [comment]
This is part of a larger, fundamental issue within the EIMS database, insofar that there is no column to join processed CTD data against any other sample collection with an `event_pk`. Ideally, this should be addressed at the database level. @finnshort will bring this up with @fostermh in their next meeting.

**Comment by JessyBarrette** [comment]
Right, my comment was mostly referring to the active work Brent is doing right now regarding the matching of the bottle with the CTDs. I've been helping him with developing an algorithm to do that job.

**Comment by fostermh** [comment]
Also, number 2 on finn's development list is to create a better link
between ctd data and the eims samples. Keep Finn and my self in the loop as
the algorithm develops as we may be able to speed up the linking process in
the database using the work you all are doing. Long term we want ctd cast
data directly linked to sampling events so this whole process is much
easier.


Matthew Foster
Hakai Institute, Calvert Island, BC, Canada / www.hakai.org
Tula Foundation, Quadra Island, BC, Canada / www.tula.org
